src/main/java/mate/academy/hwspringbootintro/repository/spec/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.hwspringbootintro.repository.spec;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.hwspringbootintro.dto.BookSearchParameters;
+import mate.academy.hwspringbootintro.model.Book;
+import mate.academy.hwspringbootintro.repository.SpecificationBuilder;
+import mate.academy.hwspringbootintro.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String AUTHOR_PROVIDER_KEY = "author";
+    private static final String TITLE_PROVIDER_KEY = "title";
+    private static final String ISBN_PROVIDER_KEY = "isbn";
```

Comment:
Better make it public and reuse in your providers and have only one place for the future changes.